### DISCUSSION TITLE
[4.x]: Custom default address formatter for the CP

### DISCUSSION THREAD

**Discussion by rob-c-baker** [craftcms/cms#12673] [Ideas] [answered]
See: https://github.com/craftcms/cms/issues/12644

There is no way to inject a custom formatter for the address displayed as a result of the HTML produced by `craft\helpers\Cp::addressCardHtml()`. This mean plugins that use this method cannot use custom address formatters.

It would be great if another parameter was added, or a key provisioned in the passed in `$config` array to pass a customised address formatter to the method (Which then gets passed to `craft\services\Addresses::formatAddress()`.

An additional event, something like `DEFINE_ADDRESS_FORMATTER` would complete the picture allowing customised address formats in the CP and front-end alike.

The https://github.com/commerceguys/addressing library does somethings that make sense in a Commerce context for addresses (only shows a sub-set of address fields for example), but we use addresses outside of this context and it would be nice to have our custom address formatter used in all visible places. Our business is wondering why data they enter into an address does not display (a) at all if the address formatter is disabled or (b) differently in different places if it is.

**Comment by bleepsandblops** [comment]
+1 for this, specifically in terms of name handling. An address with "John Alex Doe" as fullName ends up being shown in the cp as "John Doe" which surely isn't right? Why are we parsing the full name at all? Makes me think of this: https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

**Comment by brandonkelly** [comment, marked as answer]
We’ve made it possible to override the default address formatter, for Craft 4.5 (https://github.com/craftcms/cms/pull/13242).
